### ISSUE TITLE
Classifiction with targets of shape (X, 1) works, but regression does not

### ISSUE THREAD

**Issue by mitar** [automl/auto-sklearn#298]
It seems regression is stricter and it requires the shape to be `(X,)`, a vector, and does not correctly handle the shape of `(X, 1)`.

**Comment by mfeurer** [comment]
Does this fail in the target algorithm calls or when starting Auto-sklearn?

**Comment by mitar** [comment]
The problem is in fact in predict:

```python
import autosklearn.regression
import sklearn.datasets
import sklearn.model_selection
X, y = sklearn.datasets.load_boston(return_X_y=True)
y = y_train.reshape((len(y),1))
X_train, X_test, y_train, y_test = sklearn.model_selection.train_test_split(X, y, random_state=1)
automl = autosklearn.regression.AutoSklearnRegressor(time_left_for_this_task=600)
automl.fit(X_train, y_train)
automl.predict(X_test)
```

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.5/dist-packages/autosklearn/estimators.py", line 505, in predict
    X, batch_size=batch_size, n_jobs=n_jobs)
  File "/usr/local/lib/python3.5/dist-packages/autosklearn/estimators.py", line 61, in predict
    return self._automl.predict(X, batch_size=batch_size, n_jobs=n_jobs)
  File "/usr/local/lib/python3.5/dist-packages/autosklearn/estimators.py", line 61, in predict
    return self._automl.predict(X, batch_size=batch_size, n_jobs=n_jobs)
  File "/usr/local/lib/python3.5/dist-packages/autosklearn/automl.py", line 554, in predict
    predictions = self.ensemble_.predict(all_predictions)
  File "/usr/local/lib/python3.5/dist-packages/autosklearn/ensembles/ensemble_selection.py", line 195, in predict
    return np.sum(predictions, axis=0)
  File "/usr/local/lib/python3.5/dist-packages/numpy/core/fromnumeric.py", line 1814, in sum
    out=out, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/numpy/core/_methods.py", line 32, in _sum
    return umr_sum(a, axis, dtype, out, keepdims)
ValueError: could not broadcast input array from shape (127,1) into shape (127)
```

**Comment by KEggensperger** [comment]
This seems to be a problem with sklearn not handling arrays in a consistent way. 

Sklearn prefers target arrays of size [n,], but does not enforce that. For some regressors it reshapes [n,1] to [n,], but not for all.  

Having arrays of different sizes causes problems when computing an ensemble prediction, which happens only in the predict method.

Here are some examples (sklearn==0.18.1)
```
import sklearn.datasets
import sklearn.linear_model
import sklearn.ensemble
import sklearn.neighbors

X, y = sklearn.datasets.load_boston(return_X_y=True)
y = y.reshape((len(y),1))
X_train, X_test, y_train, y_test = sklearn.model_selection.train_test_split(X, y, random_state=1)

# RF and ARD output warning and reshape array
rf = sklearn.ensemble.RandomForestRegressor()
rf.fit(X_train, y_train)
print("RF", rf.predict(X_train).shape)

ard = sklearn.linear_model.ARDRegression()
ard.fit(X_train, y_train)
print("ARD", ard.predict(X_train).shape)

# KNN and RR output array with the same shape
knn = sklearn.neighbors.KNeighborsRegressor()
knn.fit(X_train, y_train)
print("KNN", knn.predict(X_train).shape)

rr = sklearn.linear_model.Ridge()
rr.fit(X_train, y_train)
print("RR", rr.predict(X_train).shape)
```

One solution would be to also throw a warning, when the user passes a [n,1] array for regression. Or manually change arrays shapes in the respective regressors.

**Comment by mfeurer** [comment]
This is potentially related to #301. We're working on a fix.

**Comment by mfeurer** [comment]
Will be fixed in the upcoming release via #368.
